### ISSUE TITLE
Displaying date on CartesianTimeAxis

### ISSUE THREAD

**Issue by niekvanderkooy** [pepstock-org/Charba#4]
When drawing a `CartesianTimeAxis`, is it possible to show the date (not just the time) for each tick? Or more generally, is it possible to set the format of how the ticks are displayed?

For the tooltip this is possible through `((CartesianTimeAxis) axis).getTime().setTooltipFormat()`. I have also found `axis.getTicks.setCallback(TickCallback)`, but this only seems to work for a `CartesianLinearAxis` - it has no effect on a `CartesianTimeAxis`

**Comment by stockiNail** [comment]
@niekvanderkooy TimeSeries support in CHART.JS (and then in CHARBA as well) is provided by Moment.js. [Here](http://www.chartjs.org/docs/latest/axes/cartesian/time.html#display-formats) Chart.Js doc.

To set your display format, you should use something like that:

```java
axis.getTime.getDisplayFormats().setDisplayFormat()setDisplayFormat(TimeUnit.day, "MMM D h:mm a");
axis.getTime().setUnit(TimeUnit.day);
```
This method can override the defaults:

Name | Default | Example
-- | -- | --
millisecond | 'h:mm:ss.SSS a' | 11:20:01.123 AM
second | 'h:mm:ss a' | 11:20:01 AM
minute | 'h:mm a' | 11:20 AM
hour | 'hA' | 11AM
day | 'MMM D' | Sep 4
week | 'll' | Sep 4 2015
month | 'MMM YYYY' | Sep 2015
quarter | '[Q]Q - YYYY' | Q3 - 2015
year | 'YYYY' | 2015

Here is the formats syntax:

Input | Example | Description
-- | -- | --
gggg | 2014 | Locale 4 digit week year
gg | 14 | Locale 2 digit week year
w ww | 1..53 | Locale week of year
e | 0..6 | Locale day of week
ddd dddd | Mon...Sunday | Day name in locale set by moment.locale()
GGGG | 2014 | ISO 4 digit week year
GG | 14 | ISO 2 digit week year
W WW | 1..53 | ISO week of year
E | 1..7 | ISO day of week
H HH | 0..23 | Hours (24 hour time)
h hh | 1..12 | Hours (12 hour time used with a A.)
k kk | 1..24 | Hours (24 hour time from 1 to 24)
a A | am pm | Post or ante meridiem (Note the one character a p are also considered valid)
m mm | 0..59 | Minutes
s ss | 0..59 | Seconds
S SS SSS | 0..999 | Fractional seconds
Z ZZ | +12:00 | Offset from UTC as +-HH:mm, +-HHmm, or Z

**Comment by niekvanderkooy** [comment]
@stockiNail Thanks for your answer. However, I tried this and it does not seem to work for me.
I have the following example:
```
        ScatterChart chart = new ScatterChart();

        chart.getOptions().setShowLines(true);

        ScatterDataset dataset = new ScatterDataset();

        DataPoint dp1 = new DataPoint();
        dp1.setT(new Date(0)); dp1.setY(20);

        DataPoint dp2 = new DataPoint();
        dp2.setT(new Date(100000)); dp2.setY(30);

        DataPoint dp3 = new DataPoint();
        dp3.setT(new Date(200000)); dp3.setY(40);

        dataset.setDataPoints(dp1, dp2, dp3);

        chart.getData().setDatasets(dataset);

        Axis axis = new CartesianTimeAxis(chart);
        ((CartesianTimeAxis) axis).getTime().getDisplayFormats().setDisplayFormat(TimeUnit.second, "HH:mm:ss");
        ((CartesianTimeAxis) axis).getTime().setUnit(TimeUnit.second);

        chart.getOptions().getScales().setXAxes(axis);

        return chart;
```
but I still get the time in the default format of `h:mm:ss a` instead of `HH:mm:ss`. Can you see if there's anything wrong with my code?

**Comment by stockiNail** [comment]
@niekvanderkooy thank you very much! I found a bug into [Time](https://github.com/pepstock-org/Charba/blob/master/src/org/pepstock/charba/client/options/scales/Time.java) class. Added row 65, 
[commit](https://github.com/pepstock-org/Charba/commit/c91f56127279b0ddbab575ce15d4df45aa21b2f2).

The DisplayFormats object is not inserted into Javascript object Time.

Fixed into master. 

![scattertime](https://user-images.githubusercontent.com/11741250/40114518-836622e4-590d-11e8-9db2-e078a543e9af.png)

**Comment by niekvanderkooy** [comment]
Cool, thank you very much!
Would it maybe be possible to do a 1.3.1 bugfix release? Since I'm pulling in Charba using Maven.

**Comment by stockiNail** [comment]
Yes, I can. I need a couple of days to do the whole procedure :(
If you agree, we will close this issue only when you will be able to test in application and we will have the confirmation that it works.

**Comment by niekvanderkooy** [comment]
Yep, that's fine! Thanks for all the effort

**Comment by niekvanderkooy** [comment]
I can confirm that this issue is fixed in v1.4!

**Comment by stockiNail** [comment]
@niekvanderkooy the bug has been fixed into version [1.4](https://github.com/pepstock-org/Charba/releases/tag/1.4). You can also get Charba from [MVN](https://mvnrepository.com/artifact/org.pepstock/charba/1.4). Thank you
